// containers/login/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {fetchLogin} from 'actions/user'

class Login extends Component {

    constructor(props) {
        super(props)
        this.state = {
            login: '',
            password: '',
        }
    }

    handlerLoginChange = (e) => {
        this.setState({login: e.target.value})


    }
    handlerPassChange = (e) => {
        this.setState({password: e.target.value})

    }
    handlerOnClick = (e) => {
        e.preventDefault()
        this.props.fetchLogin(this.state.login, this.state.password)
        this.clearPass()

    }
    checkStatusLogin=()=>{
        const {status, data} = this.props.users
        localStorage.setItem('login', 'false')
        if (status) {
            if (status === 'ok') {
                const {history} = this.props
                localStorage.setItem('login', 'true')
                localStorage.setItem('id', data.id)
                history.push('/profile')
            }
            else {
                return <p>Имя пользователя или пароль введены не верно.</p>
            }
        }
        return <p></p>
    }

    clearPass=()=>{
        this.setState({password:''})
    }

    checkButtonSend=()=>{
        if((this.state.login !=='')&&(this.state.password !==''))
            return ''
        else
            return 'false'
    }
    render(){

        return(

            <div className="">

                <form className="form-group" onSubmit={this.handlerOnClick}>

                        <input
                            type="email"
                            placeholder="Enter email"
                            className="form-control"
                            value={this.state.login}
                            onChange={this.handlerLoginChange}
                        />
                        <small className="form-text text-muted" >{this.checkStatusLogin()}</small>
                        <input
                            type="password"
                            placeholder="Password"
                            className="form-control"
                            value={this.state.password}
                            onChange={this.handlerPassChange}

                        />

                    <button className="btn btn-primary" disabled={this.checkButtonSend()}>Login</button>
                </form>
            </div>

        )
    }

}

const mapStateToProps=(state)=>{
    return {
        users: state.login
    }

}

const mapDispatchToProps = {
    fetchLogin
}


export default connect(mapStateToProps, mapDispatchToProps)(Login)
